lib/ansible/plugins/filter/encryption.py

```diff
@@ -0,0 +1,75 @@
+# Copyright: (c) 2021, Ansible Project
+
+# Make coding more python3-ish
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from ansible.errors import AnsibleFilterError, AnsibleFilterTypeError
+from ansible.module_utils._text import to_native, to_bytes
+from ansible.module_utils.six import string_types, binary_type
+from ansible.parsing.yaml.objects import AnsibleVaultEncryptedUnicode
+from ansible.parsing.vault import is_encrypted, VaultSecret, VaultLib
+from ansible.utils.display import Display
+
+display = Display()
+
+
+def do_vault(data, secret, salt=None, vaultid='filter_default', wrap_object=False):
+
+    if not isinstance(secret, (string_types, binary_type)):
+        raise AnsibleFilterTypeError("Secret passed is required to be a string, instead we got: %s" % type(secret))
+
+    if not isinstance(data, (string_types, binary_type)):
+        raise AnsibleFilterTypeError("Can only vault strings, instead we got: %s" % type(data))
+
+    vault = ''
+    vs = VaultSecret(to_bytes(secret))
+    vl = VaultLib()
+    try:
+        vault = vl.encrypt(to_bytes(data), vs, vaultid, salt)
```

Comment:
Minor nit: this has the same issue as #75131 for `salt` and `vaultid`. Undefined variables should only be fatal during templating if fail_on_undefined=True. `except Exception as e:` catches `jinja2.exceptions.UndefinedError` and raises `AnsibleFilterError` instead, which is fatal regardless.